### PR TITLE
ROCANA-10331: Process reports CPU usage percent

### PR DESCRIPTION
**Pull request by adamlamar**
Adds Windows and Linux implementations of percent CPU usage. These assume relatively frequent collection intervals, and the `Percent*` fields from `ProcTime` are not valid until `Diff()` is called. `Diff()` calculates the CPU usage over the interval between the previous reading and the current reading.

Windows' CPU usage is gathered via the WMI class `Win32_PerfFormattedData_PerfProc_Process`. Because WMI provides an instantaneous percentage of CPU usage, it is technically not necessary to call `Diff()` to compare the counters with the previous value.

Linux CPU usage is calculated by reading `/proc/stat` and `/proc/<pid>/stat`. The `Diff()` is necessary to compare against the previously collected values.

Darwin's `Diff()` method is stubbed out.

### CODE REVIEW

File: sigar_interface_test.go
Comment (matthew-dailey) [inline sigar_interface_test.go]: I guess you can't just sleep because you want the CPU to be utilized during this time?
nit: Maybe extract this all into its own function since it makes the test a little harder to read

File: sigar_linux.go
Comment (matthew-dailey) [inline sigar_linux.go]: nit: Thinking about it, I'm not sure the name "Diff" explains _why_ someone would call this method.  Maybe something like `CalculatePercentages` makes more sense?

File: sigar_linux.go
Comment (matthew-dailey) [inline sigar_linux.go:753]: This [doc](https://www.centos.org/docs/5/html/5.2/Deployment_Guide/s2-proc-stat.html) about `/proc/stat` says that time values are measured in 1/100ths of a second (jiffies).  Is that correction already done when creating the ProcTime, or does it need to be handled here?

File: sigar_interface.go
Comment (matthew-dailey) [inline sigar_interface.go]: nit: Maybe comment what time unit these `uint64`s are for.  I think it's millis?  And that the values will monotonically increase

File: sigar_interface.go
Comment (matthew-dailey) [inline sigar_interface.go]: nit: Maybe comment that these values are in the 0-100 range.  I guess that means they could also be 8-bit?  That's probably not a needed optimization

File: sigar_interface_test.go
Comment (matthew-dailey) [inline sigar_interface_test.go]: Can you make any assertions about the `Percent*` properties under Windows?

**Review by matthew-dailey — COMMENTED** [review]
This looks really good @adamlamar!  I left some nits, but also a few non-nit comments I'd like you to look over before I give an approval, thanks!

File: sigar_interface.go
Comment (matthew-dailey) [inline sigar_interface.go]: nit: Maybe comment that PageFileBytes is only seen in Windows?  (Or maybe my understanding is wrong)

File: sigar_interface_test.go
Comment (adamlamar) [reply sigar_interface_test.go]: I tried, but it was really flakey, which is why it doesn't assert. I'll add a comment to that effect.

File: sigar_linux.go
Comment (adamlamar) [reply sigar_linux.go:753]: In the `Get()` method, the `User` and `Sys` times are converted to milliseconds of used CPU time. In that sense, the conversion has already taken place, and we just use the millisecond value. That's why we divide by `diffMillis`, to get the percentage of time used vs time passed. But absolutely the value of `system.ticks` corresponds to the `HZ` value found in `asm/param.h` or via `C.sysconf(C._SC_CLK_TCK)`, and `1/HZ` is a jiffy.

The tricky part is determining the value of `HZ` - its currently hardcoded to 100, yet we may someday run on platforms with a different value.

File: sigar_interface.go
Comment (adamlamar) [reply sigar_interface.go]: Sure thing. `StartTime` is milliseconds since the epoch, the others are number of milliseconds used by the CPU. These were pre-existing in sigar, they're just part of the diff due to `gofmt`.

File: sigar_interface.go
Comment (adamlamar) [reply sigar_interface.go]: They can go above 100 if a process is using multiple threads. 8-bit is probably too low, but 32-bit should suffice. I decided to leave them at 64-bit since almost all other sigar values are 64-bit, and they will likely be cast to 64-bit for display anyway.

File: sigar_interface.go
Comment (adamlamar) [reply sigar_interface.go]: Yes, only windows is collected for now. Will comment.

File: sigar_linux.go
Comment (matthew-dailey) [reply sigar_linux.go:753]: Nice, thanks for all the details!  I'm totally cool with that value of `HZ` since it should be correct for a long time

**Review by matthew-dailey — APPROVED** [review]
LGTM @adamlamar, nice job!

**Comment by adamlamar** [comment]
Thanks for the review @matthew-dailey! Will merge when CI completes.

**Pull request merged**
